src/app/App.jsx

```diff
@@ -0,0 +1,157 @@
+   */
+  navigation: PropTypes.object.isRequired,
+  /**
+   * The root path for the site.
+   */
+  rootPath: PropTypes.string.isRequired,
+  /**
+   * The theme options the site should display in the theme utility in the toobar.
+   */
+  themes: PropTypes.object,
+  /**
+  * The locale options the site should display in the locale utility in the toobar.
+   // we should check these values are contained in i18nSupportedLocales
+   */
+  locales: PropTypes.array,
+  /**
+   * Whether or not to display the directionality utility in the toolbar.
+   */
+  hideBidiUtility: PropTypes.bool,
+  /**
+   * The directionality the site should default to. Either 'ltr' or 'rtl'.
+   */
+  defaultDir: PropTypes.string,
+  /**
+   * The theme the site should default to.
+   */
+  defaultTheme: PropTypes.string,
+  /**
+   * Injected by react-routed: represent where the app is now, where you want it to go,
+   * or even where it was.
+   */
+  location: PropTypes.shape({
+    pathname: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  appTitle: '',
+  appSubtitle: '',
+  appLogoSrc: undefined,
+  navigation: undefined,
+  hideBidiUtility: false,
+  defaultDir: 'ltr',
+  defaultTheme: 'Default Theme',
+  themes: {
+    'Default Theme': '',
+  },
+  locales: i18nSupportedLocales,
+  location: undefined,
+};
+
+const locale = document.getElementsByTagName('html')[0].getAttribute('lang');
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dir: props.defaultDir,
+      locale,
+      theme: props.defaultTheme,
+    };
+    this.handleBidiChange = this.handleBidiChange.bind(this);
+    this.handleThemeChange = this.handleThemeChange.bind(this);
+    this.handleLocaleChange = this.handleLocaleChange.bind(this);
+  }
+
+  handleBidiChange(e) {
+    document.getElementsByTagName('html')[0].setAttribute('dir', e.currentTarget.id);
+    this.setState({ dir: e.currentTarget.id });
+  }
+
+  handleLocaleChange(e) {
+    this.setState({ locale: e.currentTarget.id });
+  }
+
+  handleThemeChange(e) {
+    this.setState({ theme: e.currentTarget.id });
+  }
+
+  render() {
+    let appLogo;
+    if (this.props.appLogoSrc) {
+      appLogo = (<Image variant="rounded" src={this.props.appLogoSrc} height="26px" width="26px" isFluid />);
+    }
+
+    const applicationHeader = (
+      <ApplicationHeader
+        title={this.props.appTitle}
+        subtitle={this.props.appSubtitle}
+        logo={appLogo}
+        locale={this.state.locale}
+        locales={this.props.locales}
+        onLocaleChange={this.handleLocaleChange}
+        hideBidiUtility={this.props.hideBidiUtility}
+        dir={this.state.dir}
+        onDirChange={this.handleBidiChange}
+        theme={this.state.theme}
+        themes={Object.keys(this.props.themes)}
+        onThemeChange={this.handleThemeChange}
+        navigation={matchPath(this.props.location.pathname, this.props.rootPath) ? this.props.navigation : null}
```

Comment:
@mjhenkes @bjankord I updated this rendering so the `#/tests/` route now maintains the utilities bar with no navigation